controllers/applicationvolumereplication_controller.go

```diff
@@ -195,295 +191,471 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 		return ctrl.Result{}, errorswrapper.Wrap(err, "failed to get AVR object")
 	}
 
-	subscriptionList := &subv1.SubscriptionList{}
-	listOptions := []client.ListOption{
-		client.InNamespace(avr.Namespace),
-		client.MatchingLabels(avr.Spec.SubscriptionSelector.MatchLabels),
+	avrPlRule, userPlRule, err := r.getPlacementRules(ctx, avr)
+	if err != nil {
+		logger.Error(err, "failed to get PlacementRules")
+
+		return ctrl.Result{}, err
 	}
 
-	err = r.Client.List(ctx, subscriptionList, listOptions...)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			logger.Error(err, "failed to find subscription list", "namespace", avr.Namespace)
+	// Make sure that we give time to the cloned PlacementRule to run and produces decisions
+	if len(avrPlRule.Status.Decisions) == 0 {
+		const initialWaitTime = 5
 
-			return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{RequeueAfter: time.Second * initialWaitTime}, nil
+	}
+
+	// check if this is the initial deployement.  0 decision indicates that this is
+	// the first time using this placement rule.
+	if len(userPlRule.Status.Decisions) == 0 {
```

Comment:
It shouldn't.  Or let me ask, which one would be nil, the status?  The Status shouldn't, it is a member by value, but the Decisions can be, but `len` is smart enough to return 0 on a nil object

Reply:
I shouldn't have said `nil`. What I mean is, during failback:
- We would remove the Subscriptions (or let ACM remove the Subscriptions) from the current cluster, by setting userPl decisions as no clusters
- This is done so that, the PVCs are not in use and are also deleted, for VRG to finish reconciling them as Secondary
- Which is a pre-req before moving the workload to a peer cluster as Primary

Now in between all this, if reconcile does not complete, and we get called again, we would set the userPl cluster list here. Which should not be done if in the middle of failback.

Reply:
We need to discuss. The way I see it is that once the userPlRule.Status.Decisions is NOT empty, then it will remain NOT empty all the time.  I don't see a window where we change it to empty then none empty.  The deletion of the subscriptions should be done once we update the UserPlacementRule to a different cluster. ACM should delete subs from one cluster and populate another.

Reply:
(updating based on discussion)
- Option A: UserPl changes from one to the other cluster
    - AVR still needs to ensure VRG reaches Secondary on source cluster before creating it as Primary on the destination cluster
    - OCM may roll out Subscriptions, and hence PVCs and these will bind to restored PVs on the destination cluster
    - The unknown is, if mount of these PVCs will fail in the destination cluster as pods are brought up
- Option B: UserPl moves to empty cluster list, and then once source is cleaned up, changes to destination cluster
    - Avoids the unknown and also Subscriptions wasting cycles on the destination for VRG to be ready

The decision was to go with Option B